src/features/defi/providers/osmosis/components/OsmosisManager/Lp/Overview/OsmosisLpOverview.tsx

```diff
@@ -92,93 +81,85 @@ export const OsmosisLpOverview: React.FC<OsmosisOverviewProps> = ({
   )
 
-  const calculateBalances = useCallback(
-    async (
-      lpAsset: Asset,
-      lpAssetBalanceBaseUnit: string,
-    ): Promise<opportunityBalances | undefined> => {
-      if (!(osmosisOpportunity && underlyingAsset0 && underlyingAsset1)) return undefined
-
-      const id = getPoolIdFromAssetReference(fromAssetId(lpAsset.assetId).assetReference)
-      if (!id) return undefined
-      const poolData = await getPool(id)
-
-      if (
-        !(
-          poolData &&
-          poolData.pool_assets &&
-          poolData.total_shares &&
-          poolData.total_weight !== '0'
-        )
-      ) {
-        return undefined
-      }
-
-      const poolOwnershipFraction = bnOrZero(lpAssetBalanceBaseUnit)
-        .dividedBy(bnOrZero(poolData.total_shares.amount))
-        .toString()
-
-      const underlyingAsset0AllocationPercentage = bnOrZero(poolData.pool_assets[0].weight)
-        .dividedBy(bnOrZero(poolData.total_weight))
-        .toString()
-      const underlyingAsset1AllocationPercentage = bnOrZero(poolData.pool_assets[1].weight)
-        .dividedBy(bnOrZero(poolData.total_weight))
-        .toString()
-
-      const underlyingAsset0Balance = bnOrZero(poolOwnershipFraction)
-        .multipliedBy(poolData.pool_assets[0].token.amount)
-        .toString()
-      const underlyingAsset1Balance = bnOrZero(poolOwnershipFraction)
-        .multipliedBy(poolData.pool_assets[1].token.amount)
-        .toString()
-
-      const underlyingAsset0CryptoBalancePrecision = bnOrZero(underlyingAsset0Balance)
-        .dividedBy(bn(10).pow(bnOrZero(underlyingAsset0?.precision)))
-        .toFixed(2)
-        .toString()
-
-      const underlyingAsset1CryptoBalancePrecision = bnOrZero(underlyingAsset1Balance)
-        .dividedBy(bn(10).pow(bnOrZero(underlyingAsset1?.precision)))
-        .toFixed(2)
-        .toString()
-
-      return {
-        underlyingAssetBalances: [
-          {
-            ...underlyingAsset0,
-            allocationPercentage: underlyingAsset0AllocationPercentage,
-            cryptoBalancePrecision: underlyingAsset0CryptoBalancePrecision,
-          },
-          {
-            ...underlyingAsset1,
-            allocationPercentage: underlyingAsset1AllocationPercentage,
-            cryptoBalancePrecision: underlyingAsset1CryptoBalancePrecision,
-          },
-        ],
-        fiatBalance: bnOrZero(poolOwnershipFraction)
-          .multipliedBy(bnOrZero(osmosisOpportunity.tvl))
-          .toString(),
-      }
-    },
-    [osmosisOpportunity, underlyingAsset0, underlyingAsset1],
-  )
-
   useEffect(() => {
-    if (!lpAsset) return
-    if (opportunityBalances) return
     ;(async () => {
-      const balances = await calculateBalances(lpAsset, lpAssetBalance)
-      setOpportunityBalances(balances)
+      const id = getPoolIdFromAssetReference(fromAssetId(lpAssetId).assetReference)
+      if (!id) return
+      const poolData = await getPool(id)
+      setOsmosisPool(poolData)
```

Comment:
happy to approve this particular PR, but don't like we're missing a bunch of normal patterns we'd use here

a) views should not be responsible for fetching data
b) should have caip adapters for `lpAssetId` -> `poolId`
c) we're passing vendor ids (`poolId`) around internally
d) stuff like this is responsible for overfetching

Reply:
@0xdef1cafe 

a) Happy to fix if you show me where I should move the call. We need up-to-date price data here to do the LP token -> underlying asset calculations correctly, so I figured I'd fetch once on component load. I agree that it's nicer to move any data-fetching code out of the view layer.

b) Good idea, can do.

Reply:
What about adding this to LP metadata resolvers? Since `getPool` effectively returns Osmosis pool-specific metadata, we could add a new `OsmosisLpSpecificMetadata` union (see the current `OpportunityMetadata` union) so we ensure this is fetched at the right time, from the right place, and not over-fetched